crates/ty_python_semantic/resources/mdtest/ty_extensions.md

```diff
@@ -491,4 +497,6 @@ def _(
     reveal_type(c4)  # revealed: (...) -> Foo
 
     reveal_type(c5)  #  revealed: (x: int) -> str
+
+    reveal_type(c6)  # revealed: (x: int) -> Foo
```

Comment:
This would previously reveal `(x: int) -> bound method Foo.returns_self(x: int) -> Foo` instead of `(x: int) -> Foo` (https://play.ty.dev/f4b4b1a6-ae47-4338-9be4-07134afc237a)